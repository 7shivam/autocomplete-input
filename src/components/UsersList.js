import React from "react";

const UsersList = ({
  isListVisible,
  users,
  currentIndex,
  onSelect
}) => {
  return (
    <div className={`dropdown ${isListVisible ? "is-active" : ""} `}>
      <div className="dropdown-menu" id="dropdown-menu">
        <div className="dropdown-content">
          {users.map((user, index) => (
            <a
              key={index}
              onClick={() => onSelect(user)}
              className={`dropdown-item ${
                currentIndex === index ? "is-active" : ""
              }`}
            >
              {user}
            </a>
          ))}
        </div>
      </div>
    </div>
  );
};

export default UsersList;
