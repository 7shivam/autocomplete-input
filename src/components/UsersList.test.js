import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import UsersList from './UsersList';

test('renders user input', () => {
    const tree = renderer
        .create(
            <UsersList
                users={[]}
                onSelect={() => {}}
                isListVisible={true}
                currentIndex={0}
            />
        ).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('renders user input when users list is provided', () => {
    const tree = renderer
        .create(
            <UsersList
                users={[
                    'Manoj Singh Negi',
                    'Ross Geller'
                ]}
                onSelect={() => {}}
                isListVisible={true}
                currentIndex={0}
            />
        ).toJSON();
    
    expect(tree).toMatchSnapshot();
})


test('renders user input when isListVisible is false', () => {
    const tree = renderer
        .create(
            <UsersList
                users={[
                    'Manoj Singh Negi',
                    'Ross Geller'
                ]}
                onSelect={() => {}}
                isListVisible={false}
                currentIndex={0}
            />
        ).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('call onSelect when clicked on any item', () => {
    const onSelect = jest.fn()

    render(
        <UsersList
            users={[
                'Manoj Singh Negi',
                'Ross Geller'
            ]}
            onSelect={onSelect}
            isListVisible={true}
            currentIndex={0}
        />
    )

    userEvent.click(screen.getByText('Manoj Singh Negi'))
    expect(onSelect).toHaveBeenCalledWith('Manoj Singh Negi')
})
