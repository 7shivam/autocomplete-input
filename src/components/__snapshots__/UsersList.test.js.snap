// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user input 1`] = `
<div
  className="dropdown is-active "
>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
  >
    <div
      className="dropdown-content"
    />
  </div>
</div>
`;

exports[`renders user input when isListVisible is false 1`] = `
<div
  className="dropdown  "
>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
  >
    <div
      className="dropdown-content"
    >
      <a
        className="dropdown-item is-active"
        onClick={[Function]}
      >
        Manoj Singh Negi
      </a>
      <a
        className="dropdown-item "
        onClick={[Function]}
      >
        Ross Geller
      </a>
    </div>
  </div>
</div>
`;

exports[`renders user input when users list is provided 1`] = `
<div
  className="dropdown is-active "
>
  <div
    className="dropdown-menu"
    id="dropdown-menu"
  >
    <div
      className="dropdown-content"
    >
      <a
        className="dropdown-item is-active"
        onClick={[Function]}
      >
        Manoj Singh Negi
      </a>
      <a
        className="dropdown-item "
        onClick={[Function]}
      >
        Ross Geller
      </a>
    </div>
  </div>
</div>
`;
